### ISSUE TITLE
Question about the loss calculated

### ISSUE THREAD

**Issue by ChristineSeven** [tianyi-lab/Cherry_LLM#25]
I  noticed how your loss calculated, 
`    with torch.no_grad():
        outputs = model(input_ids, labels=labels)

    loss = outputs.loss
    perplexity = torch.exp(loss)

    losses = []
    logits = outputs.logits
    for i in range(1, end_token):
        log_prob_dist = log_softmax(logits[0, i-1])
        true_token = input_ids[0, i]
        token_loss = nll_loss(log_prob_dist.unsqueeze(0), true_token.unsqueeze(0))
        losses.append(token_loss.item())
`
then do `losses.mean`, so why not directly use loss, is there any difference? Apprecicate your reply!

**Comment by MingLiiii** [comment]
Thanks for your interest! 
They are the same. Please refer to https://github.com/tianyi-lab/Cherry_LLM/issues/3.

**Comment by ChristineSeven** [comment]
Thanks！
